dietpi/dietpi-software

```diff
@@ -4199,6 +4205,111 @@ _EOF_
+			# - Start with fresh instance, to allow clean pyenv and Python updates and fix broken instances.
+			[[ -d "$micro_pyenv_dir" ]] && G_EXEC rm -R "$micro_pyenv_dir"
+			G_EXEC mv pyenv-master "$micro_pyenv_dir"
+			G_EXEC chown -R dietpi:dietpi "$micro_pyenv_dir"
+
+			# 2. Generate script to activate pyenv
+			echo "#!/bin/dash
+cd $micro_pyenv_dir
+export PYENV_ROOT='$micro_pyenv_dir'
+export PATH=\"\$PYENV_ROOT/bin:\$PATH\"
+eval \"\$(pyenv init --path)\"
+eval \"\$(pyenv init -)\"
+cd $micro_dir
+PS1=\"(microblog.pub) \$PS1\" " > "$micro_pyenv_activate_file"
+			G_EXEC chown dietpi:dietpi "$micro_pyenv_activate_file"
+
+			## Install Python 3.10 and Poetry
+			G_EXEC_DESC='Installing Python with into pyenv'
+			G_EXEC_OUTPUT=1 G_EXEC sudo -u dietpi dash -c "
+source $micro_pyenv_activate_file
+pyenv install $micro_python_version
+pyenv local $micro_python_version
+pip3 install --no-cache-dir -U pip setuptools wheel
+exec pip3 install --no-cache-dir poetry"
+
+			G_EXEC_DESC='Installing Poetry dependencies (can take a while)'
+			G_EXEC_OUTPUT=0 G_EXEC sudo -u dietpi dash -c "
+source $micro_pyenv_activate_file
+pyenv local $micro_python_version
+poetry install
+## Skip these for now:
+## - poetry run inv configuration-wizard
+## - poetry run inv migrate-db
+poetry env info | grep -oP '(?<=^Path:\s).*/virtualenvs/.*' | xargs echo > $micro_virtualenv
+"
+			if [[ $(cat "$micro_virtualenv") = "" ]]; then
+				echo "Unable to locate virtualenv" && exit 255
+			fi
+			## 3. Generate functions for later configuration and install to profile
+			cat << _EOF_ > "$micro_functions"
+#!/bin/bash
+
+microblog_pub (){
+	case \$1 in
+		v*) bash --init-file "$micro_pyenv_activate_file";;  ## venv
+		c*) bash -c "source $micro_pyenv_activate_file; poetry run inv configuration-wizard && poetry run inv migrate-db";; ## configure
+		u*) bash -c "source $micro_pyenv_activate_file; git pull && poetry run inv update";; ## update
+		*) echo "Micropub helper script
+
+    usage:  microblog_pub  <venv|configure|update>
+
+where:
+   configure   runs the configuration wizard and updates the microblog environment.
+   update      pulls the latest changes from the microblog repository and reconfigures.
+   venv        starts a virtual environment where the microblog can be configured.
+" && return;;
+	esac
+
+	sudo systemctl restart microblog-pub
+
+}
+
+_EOF_
+			## 4. Generate systemd script
+			cat << _EOF_ > "$micro_systemd"
+[Unit]
+Description=Microblog.pub Web Service
+Wants=network-online.target
+After=network-online.target
+
+[Service]
+User=dietpi
+SyslogIdentifier=Microblog.pub
+ExecStart=bash -c "source $micro_pyenv_activate_file; poetry run supervisord -c misc/supervisord.conf -n"
+WorkingDirectory=$micro_dir
+Environment="VENV_DIR=$(cat "$micro_virtualenv")"
+RestartForceExitStatus=100
+
+[Install]
+WantedBy=multi-user.target
+_EOF_
+
+			G_EXEC systemctl enable --now microblog-pub
```

Comment:
```suggestion
```
Add the service to https://github.com/MichaIng/DietPi/blob/dev/dietpi/dietpi-services instead.